packages/coin-vet/src/sign.ts

```diff
@@ -0,0 +1,228 @@
+import Web3Utils from 'web3-utils';
+
+const elliptic = require('elliptic');
+const ec = new elliptic.ec('secp256k1');
+const rlp = require('rlp');
+const blake2b = require('blake2b');
+const fastJsonStableStringify = require('fast-json-stable-stringify')
+
+/**
+ * Sign VeChain Transaction
+ */
+export async function signTransaction(
+  signTxData: types.signTxType,
+  publicKey: string
+): Promise<string> {
+  const { transaction, transport, addressIndex, appPrivateKey, appId, confirmCB, authorizedCB } = signTxData;
+
+  const preActions = [];
+
+  const { script, argument } = await scriptUtil.getScriptAndArguments(addressIndex, transaction);
+
+  const sendScript = () => apdu.tx.sendScript(transport, script);
+  preActions.push(sendScript);
+  const action = () => apdu.tx.executeScript(transport, signTxData.appId, signTxData.appPrivateKey, argument);
+
+  const signature = await tx.flow.getSingleSignatureFromCoolWallet(
+    transport,
+    preActions,
+    action,
+    false,
+    confirmCB,
+    authorizedCB,
+    true
+  );
+
+  const rawTx = txUtil.getRawTx(transaction);
+  const rawData = rlp.encode(rawTx);
+ 
+  const hash = blake2b(32).update(rawData).digest('hex')
+  const data = Buffer.from(handleHex(hash), 'hex')
+  const keyPair = ec.keyFromPublic(publicKey, 'hex');
+  
+  const recoveryParam = ec.getKeyRecoveryParam(data, signature, keyPair.pub);
+  const v = recoveryParam;
+  const { r, s } = signature as {r: string, s: string};
+  rawTx.push(Buffer.concat([Buffer.from(r, 'hex'), Buffer.from(s, 'hex'), Buffer.from([v])]));
+  const serializedTx = rlp.encode(rawTx);
+  
+  return `0x${serializedTx.toString('hex')}`;
+}
+
+/**
+ * Sign VeChain Token
+ */
+ export async function signToken(
+  signTxData: types.signTxType,
+  publicKey: string
+): Promise<string> {
+  const { transaction, transport, addressIndex, appPrivateKey, appId, confirmCB, authorizedCB } = signTxData;
+
+  const preActions = [];
+
+  const { script, argument } = await scriptUtil.getVTHOScriptAndArguments(addressIndex, transaction);
+
+  const sendScript = () => apdu.tx.sendScript(transport, script);
+  preActions.push(sendScript);
+  const action = () => apdu.tx.executeScript(transport, signTxData.appId, signTxData.appPrivateKey, argument);
+
+  const signature = await tx.flow.getSingleSignatureFromCoolWallet(
+    transport,
+    preActions,
+    action,
+    false,
+    confirmCB,
+    authorizedCB,
+    true
+  );
+
+  const rawTx = txUtil.getRawTx(transaction);
+  const rawData = rlp.encode(rawTx);
+  const hash = blake2b(32).update(rawData).digest('hex')
+  const data = Buffer.from(handleHex(hash), 'hex')
+  const keyPair = ec.keyFromPublic(publicKey, 'hex');
+  
+  const recoveryParam = ec.getKeyRecoveryParam(data, signature, keyPair.pub);
+  const v = recoveryParam;
+  const { r, s } = signature as {r: string, s: string};
+  
+  rawTx.push(Buffer.concat([Buffer.from(r, 'hex'), Buffer.from(s, 'hex'), Buffer.from([v])]));
```

Comment:
```suggestion
  Buffer.concat([Buffer.from(r.padStart(64, '0'), 'hex'), Buffer.from(s.padStart(64, '0'), 'hex'), Buffer.from([v])]);
```
Sometimes `signature.s` might be in lower s form (only have 31 bytes), which is incompatible with the VeChain signature.